client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java

```diff
@@ -402,55 +396,112 @@ private NMap map2Nmap(Map<String, Object> map) throws UnsupportedOperationExcept
-        } else if (value instanceof Long) {
-            long ival = (Long)value;
-            nvalue.setIVal(ival);
-        } else if (value instanceof Float) {
-            float fval = (Float)value;
-            nvalue.setFVal(fval);
-        } else if (value instanceof Double) {
-            double dval = (Double)value;
-            nvalue.setFVal(dval);
-        } else if (value instanceof String) {
-            byte[] sval = ((String)value).getBytes();
-            nvalue.setSVal(sval);
-        } else if (value instanceof List) {
-            nvalue.setLVal(list2Nlist((List<Object>)value));
-        } else if (value instanceof Map) {
-            nvalue.setMVal(map2Nmap((Map<String, Object>)value));
-        } else if (value instanceof Value) {
-            return (Value)value;
-        } else if (value instanceof Date) {
-            nvalue.setDVal((Date)value);
-        } else if (value instanceof Time) {
-            nvalue.setTVal((Time)value);
-        } else if (value instanceof Duration) {
-            nvalue.setDuVal((Duration)value);
-        } else if (value instanceof DateTime) {
-            nvalue.setDtVal((DateTime)value);
-        } else if (value instanceof Geography) {
-            nvalue.setGgVal((Geography) value);
-        } else {
-            // unsupport other Value type, use this function carefully
-            throw new UnsupportedOperationException(
-                    "Only support convert boolean/float/int/string/map/list to nebula.Value but was"
-                    + value.getClass().getTypeName());
+        try {
+            if( value == null ) {
+                Value nValue = new Value();
+                nValue.setNVal(NullType.__NULL__);
+                return nValue;
+            }
+            Class<?> type = value.getClass();
+            Setter<Object> setter = LEAF_TYPE_AND_SETTER.get(type);
+            if( setter != null ) return setter.set( value );
+            for (Class<?> parentType: COMPLEX_TYPE_AND_SETTER.keySet()) {
+                if( isCurrentTypeOrParentType( type , parentType ) )  {
+                    return COMPLEX_TYPE_AND_SETTER.get( parentType ).set( value );
+                }
+            }
+        } catch (Exception e) {
+            throw new UnsupportedOperationException(e);
         }
-        return nvalue;
+        throw new UnsupportedOperationException(
+                "Only support convert boolean/float/int/string/map/list to nebula.Value but was"
+                        + value.getClass().getTypeName());
+    }
+
+    /**
+     * some value setter for java type (basic or nebula special type) that need convert to NValue
+     */
+    public static Map<Class<?>, Setter> LEAF_TYPE_AND_SETTER = new HashMap<Class<?>, Setter>() {{
+        put(Value.class, (Setter<Value>) (param) -> param );
+        put(boolean.class, (Setter<Boolean>) Value::bVal);
+        put(Boolean.class, (Setter<Boolean>) Value::bVal);
+        put(int.class, (Setter<Integer>) Value::iVal);
```

Comment:
is it necessary to put the base data type class? such as int.class.

Reply:
> is it necessary to put the base data type class? such as int.class.

Yes, the hashcode is different between `Integer.class` and `int.class`

Reply:
> > is it necessary to put the base data type class? such as int.class.
> 
> Yes, the hashcode is different between `Integer.class` and `int.class`

Oh... It is unnecessary... when int auto sealing, intVal.getClass() return Integer. base data type will never be visited
You are right. @Nicole00